qualtran/bloqs/basic_gates/z_basis.py

```diff
@@ -254,20 +254,17 @@ def my_tensors(
         ]
 
     def get_ctrl_system(self, ctrl_spec: 'CtrlSpec') -> Tuple['Bloq', 'AddControlledT']:
-        if ctrl_spec != CtrlSpec():
-            # Delegate to the general superclass behavior
-            return super().get_ctrl_system(ctrl_spec=ctrl_spec)
+        from qualtran.bloqs.bookkeeping import AutoPartition
+        from qualtran.bloqs.mcmt.specialized_ctrl import get_ctrl_system_1bit_cv_from_bloqs
 
-        bloq = CZ()
+        cz = CZ()
+        cz_wrapped = AutoPartition(
```

Comment:
this is kindof a regression, no? Before; you'd get a nice idiomatic CZ and now there's another level of indirection (?)

Reply:
> Btw, in the get_ctrl_sys_1bit, I return the unwrapped ctrl_bloq, so you'd only get CZ in the call graph.

can you explain this more? where does it get unwrapped? in the first return value but not when used as part of the `adder` method?

> it's kinda annoying that the target registers of Z and CZ do not match

For generality, I think it's important to keep this case supported. 

My question is: is it worth porting these gates to `get_ctrl_system_1bit_cv_from_bloqs`? Can you explicitly list what we gain? Could we keep the old code and only delegate to the new function for previously-unhandled situations? 

The `specilized_ctrl` simplifications are great for bloq authors (our "users") to make it simple to quickly support a broad set of use cases. In core parts of the library (like a Z gate) the onus is instead on the developers of the library (i.e. us) to absorb a lot of the complexity to make the users experience as straightforward as possible. To me, that means not introducing any unnecessary partitioning into the circuit